server/models/database.go

```diff
@@ -135,6 +135,17 @@ func (d *DB) AddUserVoucher(id string, voucher string) error {
 	return d.DeactivateVoucher(voucher)
 }
 
+// GetVoucherByUserID returns voucher by its user id
+func (d *DB) GetVoucherByUserID(id string) (Voucher, error) {
```

Comment:
the user can have multiple voucher requests, i think we need to find the first one that isn't approved and not used

Reply:
I believe we should, because they may end up asking for more machines afterall